src/map/mob.cpp

```diff
@@ -63,19 +63,39 @@
 // Disable this to make monsters not do any path search when looking for a target (old behavior).
 #define ACTIVEPATHSEARCH
 
-//Dynamic mob database, allows saving of memory when there's big gaps in the mob_db [Skotlex]
-//!NOTE: Mob ID is used also as index in mob db
-struct mob_db *mob_db_data[MAX_MOB_DB+1];
-struct mob_db *mob_dummy = NULL;	//Dummy mob to be returned when a non-existant one is requested.
+// Limits for the monster database
+#define MIN_MOB_DB 1000
+#define MAX_MOB_DB 3999
+#define MIN_MOB_DB2 20020
+#define MAX_MOB_DB2 31999
 
-struct mob_db *mob_db(int mob_id) { if (mob_id < 0 || mob_id > MAX_MOB_DB || mob_db_data[mob_id] == NULL) return mob_dummy; return mob_db_data[mob_id]; }
+// These define the range of available IDs for clones. [Valaris]
+#define MOB_CLONE_START MAX_MOB_DB
+#define MOB_CLONE_END MIN_MOB_DB2
+
+//Dynamic mob database
+std::map<uint16, struct mob_db> mob_db_data;
+
+struct mob_db *mob_db( int mob_id ){ 
+	try{
+		return &mob_db_data.at(mob_id);
+	}catch( std::out_of_range ){
+		return NULL;
```

Comment:
Actually we had a little discussion about that. I suggested to remove the mobdb() function completely, handle the access directly in the calling functions with mob_db_data.find() (i.e.) and try to avoid possible null pointers (or pointers at all).

Reply:
hmm why ?
I suppose is to avoid a check in the caller, but by doing this it imply the caller know the implementation of the mob_db_data. (a map a vector or whaterver).
It's probably fine cause the project is still relatively small by I prefer to hide implementation as much as possible. 
The module (mob.cpp) is the only one who should really know his db implementation the rest should just use an interface.

Reply:
I'd also prefer to hide the implementation as much as possible. Even inside the module itself, because with this we'll repeat the logic over and over again and only have duplicate code. Or have to change it if we exchange the implementation behind it.